src/bootstrap/bootstrap.py

```diff
@@ -1031,6 +1031,11 @@ def check_vendored_status(self):
                             "--sync ./src/tools/rust-analyzer/Cargo.toml " \
                             "--sync ./compiler/rustc_codegen_cranelift/Cargo.toml " \
                             "--sync ./src/bootstrap/Cargo.toml "
+
+                # get the last commit hash of the rustc directory
+                last_commit = subprocess.check_output(['git', 'rev-parse', 'HEAD']) \
+                                .decode('ascii').strip()
```

Comment:
This will not work in the following cases:

- If the last commit only exists in downstream (meaning not pushed to the upstream yet)
- If it's tarball source tree (which doesn't include `.git` in it)